### ISSUE TITLE
Chipping in

### ISSUE THREAD

**Issue by richardhp** [ib-ruby/ib-ruby#147]
I'm personally tired of Python being the de-facto scripting language when Ruby is better in almost every imaginable way.

I need to do some automation of IB for work, so I'm thinking it would be great to contribute to this project if I can, and help keep the project active and relevant so it doesn't end up getting out-dated, and then I'm forced into migrating to the Official Python version.

I had a browse through the issues and I noticed that basically every one was either very old or seemingly already resolved in some way.

So I'm wondering, is there anything I can actually do to help?  I don't have heaps of time to put in at the moment but if there's anything I can do I'd like to assist if I can.

**Comment by topofocus** [comment]
Hi Richard,
you are very welcome to participate.

I am in the process of decoupling features. The goal is to have a small, well tested tws-api program which translates tws signals into ruby messages. In addition I will eventually  provide a few service gems, to ease the usage, bind it to several databases,  etc. 

I started the ib-api gem (https://github.com/ib-ruby/ib-api). The next step there is the integration of the test suite. Then the gem is ready for a primary release. 
Then there is the challenge to adapt the code to benefit from improvements of ruby3. This might be a game changer for ruby and could inspire many python users to change. 
Any other suggestions are welcome.

**Comment by richardhp** [comment]
Yeah I did read that long issue about modularising the code, which makes sense to split it into smaller, more focused gems.

I did have a few issues running the rspec test suite.  It seemed to require a connection to the API, which made me a bit uncomfortable.  I didn't exactly want to have to use my login credentials just to test the suite.  On the other hand, this library is so tightly coupled to the underlying API, once you're done stubbing out or mocking out the underlying API you wonder what you would even be testing.  So maybe it makes sense for the Gateway to be a dependency of the test suite?

Haven't read too much about ruby 3, so I'll look into that.  Sounds exciting!

I'll jump over to the ib-api gem and see what I can do over there.  I think it's probably too early for me to start giving suggestions yet as I barely know how the code-base works etc. etc.  I'd think though, once the ib-api gem is complete, there would still be some work integrating that back into the ib-ruby gem.

There was also some mention in that long issue about decoupling other parts, for example the active record and rails dependencies, which again seemed very sensible to me.  So maybe that's another area that could be productive.

**Comment by topofocus** [comment]
To run the test, always use a demo account. Then its ok to include your credentials in the configuration files.

Is rails integration  really necessary?  
The simple-monitor-gem shows the integration into the camping suite. sinatra too should work smoothly. 
I personally depend on an integration of the orientdb database  (through active-orient gem)

**Comment by richardhp** [comment]
Ok got it on the testing.

For rails, I was just commenting on the following from the issue "The current status and future of ib-ruby project"

> ib-ruby-rails. ActiveRecord models, Rails engine

And that I thought it was a good idea to split all these features out into separate gems.  I'm not suggesting that it is a requirement though.

OrientDB looks interesting, never seen that before.  I'll have to get around to experimenting with that some time.

**Comment by topofocus** [comment]
I would add: It would be nice, to bind other schema-gems, like dry-schema, nxt_schema or parametric  to the gem.
